### PR TITLE
Feature: Add inner revert reasons for nested calls

### PR DESCRIPTION
**Pull request by ghost**
Depends on #79.

This change adds a reason to each reverted nested call, if one is provided, instead of only providing a reason for the top-level call. Also adds a test with a successful transaction that makes a failed call with a revert reason.

### CODE REVIEW

File: eth/tracers/internal/tracetest/txnOpCodeTracer_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package tracetest
```
Comment (lthibault) [inline eth/tracers/internal/tracetest/txnOpCodeTracer_test.go:1]: Can we move this to `eth/tracers/native` and rename the package to `trace_test`?  Similarly, I would recommend moving the testdata files to `eth/tracers/native/testdata`, as it's more idiomatic.

**Review by lthibault — CHANGES_REQUESTED** [review]
See inline for suggestion about directory structure, but otherwise LGTM.

(Congrats on your first PR!)

**Review by lthibault — APPROVED** [review]

File: eth/tracers/internal/tracetest/txnOpCodeTracer_test.go (hunk 1 above)
Comment (lthibault) [reply eth/tracers/internal/tracetest/txnOpCodeTracer_test.go:1]: https://blocknative.slack.com/archives/C01H2JUBQ48/p1673554679525629?thread_ts=1673553557.154409&cid=C01H2JUBQ48

**Review by alexcampbelling — APPROVED** [review]

**Comment by tyler-smith** [comment]
Rebasing onto the fixed test directory change.

**Pull request merged**